File: frontend/services/types/image_analysis.ts
/**
 * Shared TypeScript types for Food Image Analysis.
 * These mirror the backend Pydantic models exactly.
 */

export interface FoodItem {
  name: string;
  confidence: number;
  estimated_calories: number;
  portion_size: string;
}

export interface HealthierAlternative {
  original: string;
  alternative: string;
  reason: string;
  calorie_difference: number;
}

export interface ImageAnalysisResponse {
  detected_foods: FoodItem[];
  total_estimated_calories: number;
  healthier_alternatives: HealthierAlternative[];
  nutritional_notes: string;
  explanation: string;
  image_url: string | null;
}

File: backend/models/image_analysis.py
"""Pydantic models for Food Image Analysis."""

from typing import List, Optional

from pydantic import BaseModel, Field


class FoodItem(BaseModel):
    """A food item detected in an image."""

    name: str
    confidence: float = Field(..., ge=0.0, le=1.0)
    estimated_calories: int
    portion_size: str


class HealthierAlternative(BaseModel):
    """A suggested healthier food swap."""

    original: str
    alternative: str
    reason: str
    calorie_difference: int


class ImageAnalysisResponse(BaseModel):
    """Analysis result for an uploaded food image."""

    detected_foods: List[FoodItem]
    total_estimated_calories: int
    healthier_alternatives: List[HealthierAlternative]
    nutritional_notes: str
    explanation: str
    image_url: Optional[str] = None
